src/diffusers/pipelines/ip_adapter/pipeline_ip_adapter.py

```diff
@@ -0,0 +1,234 @@
+from typing import Any, Dict, List, Optional, Union
+
+import PIL
+import torch
+import torch.nn.functional as F
+from torch import nn
+from transformers import CLIPImageProcessor, CLIPVisionModelWithProjection
+
+from ...models.attention_processor import AttnProcessor, AttnProcessor2_0
+from ...utils import (
+    DIFFUSERS_CACHE,
+    HF_HUB_OFFLINE,
+    _get_model_file,
+)
+from ..pipeline_utils import DiffusionPipeline
+from .attention_processor import CNAttnProcessor, CNAttnProcessor2_0, IPAttnProcessor, IPAttnProcessor2_0
+from .image_projection import ImageProjModel
+
+
+class IPAdapterPipeline(DiffusionPipeline):
+    def __init__(
+        self,
+        pipeline: DiffusionPipeline,
```

Comment:
We don't really allow passing a `pipeline` into another pipeline. From what I can see the pipeline also only works for stable diffusion 1,2 anyways (there is only one text encoder) - could we maybe just follow the previous logic and create a `StableDiffusionIPAadapterPipeline(DiffusionPipeline)` to begin with? 

Not too keen on adding a new design paradigm here